app/views/admin/attachments/_hoc_reference_fields.html.erb

```diff
@@ -1,24 +1,62 @@
-<div class="paper-number clearfix">
-  <%= form.text_field :hoc_paper_number,
-                      label_text: "House of Commons paper number",
-                      class: 'input-md-3' %>
+<%
+  if form.object.unnumbered_hoc_paper.eql?(false)
+    numbered_hoc_paper = true
+    unnumbered_hoc_paper = false
+  elsif form.object.unnumbered_hoc_paper.eql?(true)
+    numbered_hoc_paper = false
+    unnumbered_hoc_paper = true
+  else
+    numbered_hoc_paper = false
+    unnumbered_hoc_paper = false
+  end
+%>
 
-  <%= form.check_box :unnumbered_hoc_paper,
-                     label_text: "Unnumbered act paper" %>
-
-  <p class="help-block">
-    Fill in the command and House of Commons boxes to publish an official
-    document which will appear in the list of official documents.
-  </p>
+<div class="govuk-!-margin-bottom-8">
+  <%= render "govuk_publishing_components/components/radio", {
+    heading: "House of Commons paper number",
+    heading_level: 2,
+    heading_size: "l",
+    name: "attachment[unnumbered_hoc_paper]",
+    id_prefix: "hoc_paper_number_conditional",
+    items: [
+      {
+        value: "0",
+        text: "Numbered",
+        checked: numbered_hoc_paper,
```

Comment:
I think you can just put this in:

```suggestion
        checked: form.object.numbered_hoc_paper,
```

Reply:
Hmm, I tried some variations on that but it didn't persist the state of the radio when the user submits the form with errors. The problem is that `numbered_hoc_paper` is not a valid thing though `unnumbered_hoc_paper` is. I am aware though that setting the values the way it is done here looks pretty clunky and I'm sure there's a more elegant way to deal with it. I just couldn't see what it would be.

Reply:
I have now updated this to something that seems to render as required and is somewhat more elegant.